enhancements/update/duration-of-migration-to-multi-arch.md

```diff
@@ -0,0 +1,166 @@
+---
+title: duration-of-migration-to-multi-arch
+authors:
+  - "@hongkailiu"
+reviewers:
+  - "@wking"
+approvers:
+  - "@PratikMahajan"
+api-approvers:
+  - "None"
+creation-date: 2024-10-01
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OTA-962
+see-also:
+  
+replaces:
+
+superseded-by:
+
+---
+
+# Duration of Migration To Multi Arch
+
+## Summary
+
+This enhancement provides a proposal to improve Cluster Version Operation's evaluation on the progress of [migrating an existing single-architecture cluster to multi-architecture](https://docs.openshift.com/container-platform/4.17/updating/updating_a_cluster/migrating-to-multi-payload.html).
+
+
+## Motivation
+
+During a cluster upgrade, the Cluster Version Operator (CVO) downloads the desired release payload and applies the manifests of OpenShift operators to the cluster.
+Each OpenShift operator reports the versions of its own and its operands via `ClusterOperator.status.versions` so that CVO can oversee the upgrade process of the operator.
+When all of their versions are the desired version, CVO considers that the upgrade of the operator completes.
+
+The command `oc adm upgrade --to-multi-arch` is used to migrate a cluster from a single-architecture to a multi-architecture cluster. It is implemented via a cluster upgrade by switching the underlying images to the ones from the multi-arch payload *at the current version*.
+Because `ClusterOperator.status.versions` contains the information only about the operator's version number at the moment and `--to-multi-arch` starts the upgrade targeting *the same version*, the upgrade of each operator finishes right away from the CVO's point of view. Without the ability to recognize the component's turning into multi-arch, CVO's evaluation of the upgrade progress is inaccurate, i.e., the operators' upgrade is still ongoing when CVO claims complete. This enhancement is to define a proposal to address the issue.
+
+
+### User Stories
+
+* As a cluster admin, I want CVO to claim that migrating a cluster to multi-architecture is complete correctly.
+
+
+### Goals
+
+This enhancement is to provide a proposal that addresses the issue about the upgrade progress during a cluster migration to multi-arch. The proposal has impact only after the migration to multi-arch starts.
+
+### Non-Goals
+
+Whether or not 'oc adm upgrade' is executed succesfully or the requested upgrade is accepted is not in the scope of this enhancement. So it is _not_ required to have test cases for this enhancement to cover (although they are valid in general):
+- upgrade from multi-arch to single-arch,
```

Comment:
I know this is a non-goal, but I don't believe upgrade from multi-arch to single-arch is a valid pathway at all. As far as I know, once you upgrade to multi-arch, you stay at multi-arch.

Reply:
multi-arch to single arch is not a supported path, yes although it can be forced if the needed.

Reply:
This is to address [the comment](https://github.com/openshift/enhancements/pull/1690#discussion_r1796607120).
They are valid cases in the sense the cases can be carried out. I do not mean they are supported paths.
I would image that migration to multi-arch on an already multi-arch cluster is a no-op action.
As it is a no-goal, I do not include the expecting results for those cases.